### DISCUSSION TITLE
Convert every layer/file/image in a PSD file into JXL files

### DISCUSSION THREAD

**Discussion by a84r7a3rga76fg** [ImageMagick/ImageMagick#6645] [Help]
### ImageMagick version

7.1.1-15

### Operating system, version and so on

Windows 10

### Description

Does anyone know how to do that while keeping transparency and alpha channels?

**Comment by fmw42** [comment]
Post and example PSD file

**Reply by a84r7a3rga76fg** [reply]
https://github.com/ImageMagick/ImageMagick/discussions/6477#discussioncomment-6423332

**Reply by fmw42** [reply]
That image has only one layer and no transparency, only one alpha channel.  Please provide an example that has more than one layer and transparency and and alpha channel.

**Reply by fmw42** [reply]
Does JXL support both transparency and alpha channels?  If not, what do you want to happen to them?

**Comment by snibgo** [comment]
JXL can record up to 4099 channels. (See https://en.wikipedia.org/wiki/JPEG_XL)

However, from jxl.c function WriteJXLImage(), it seems IM can't write more than RGB or RGBA channels.
